lnwallet/btcwallet/psbt.go

```diff
@@ -59,24 +63,43 @@ func (b *BtcWallet) FundPsbt(packet *psbt.Packet, minConfs int32,
 		keyScope   *waddrmgr.KeyScope
 		accountNum uint32
 	)
+
+	// No change key scope should have been specified for custom accounts.
+	if changeScope != nil &&
```

Comment:
Now that there is some duplication in this code anyway, maybe rewriting it like this is more readable:

```golang
	switch accountName {
	// If the default/imported account name was specified, we'll provide a
	// nil key scope to FundPsbt, allowing it to select inputs from all
	// scopes (NP2WKH, P2WKH, P2TR). By default, the change key scope for
	// these accounts will be P2WKH.
	case lnwallet.DefaultAccountName:
		if changeScope == nil {
			changeScope = &waddrmgr.KeyScopeBIP0084
		}

		accountNum = defaultAccount

	case waddrmgr.ImportedAddrAccountName:
		if changeScope == nil {
			changeScope = &waddrmgr.KeyScopeBIP0084
		}

		accountNum = importedAccount

	// Otherwise, map the account name to its key scope and internal account
	// number to only select inputs from said account. The change key scope
	// will always be the same as the coin selection.
	default:
		if changeScope != nil {
			return 0, fmt.Errorf("couldn't specify a change address " +
				"type for custom accounts")
		}

		scope, account, err := b.wallet.LookupAccount(accountName)
		if err != nil {
			return 0, err
		}
		keyScope = &scope
		changeScope = keyScope
		accountNum = account
	}
```

Reply:
Indeed, it improves readability!